csrc/scheduler/reduction.cpp

```diff
@@ -637,6 +680,21 @@ std::shared_ptr<ReductionParams> outerReductionHeuristic(
   // Start bdimx as a warp
   bdimx = std::min(min_warp_size, total_iteration_numel);
 
+  // Set iter_unroll_factor (vectorization) before increasing bdimx.
+  // for block reduce: may be further decreased to use more blocks.
+  // for grid reduce: this is the final value.
+  iter_unroll_factor = [&]() {
```

Comment:
I was trying to group the code used to set `iter_unroll_factor` into a code block. This immediately-invoked lambda separate tmp local variables from other variables. Easy to extend when we want to change how `iter_unroll_factor` is set.
maybe a little bit overkill since I simplified the process to set `iter_unroll_factor`. 

But when set  `is_grid_reduce` this can help in maintaining clarity and avoiding polluting the surrounding scope with temporary variables (tmp_bdimy, tmp_redu_unroll, thread_local_redu) that are only relevant to the initialization process.

Reply:
nitpick on coding style.

If it's just a code block we want, we can just use a code block.
```
{
    // ...
    iter_unroll_factor = scheduler_utils::lastPow2(factor);
}
```